src/configobj/__init__.py

```diff
@@ -1210,8 +1215,8 @@ def _load(self, infile, configspec):
         except AttributeError:
             pass
 
-        if isinstance(infile, six.string_types):
-            self.filename = infile
+        if isinstance(infile, (six.string_types, Path)):
+            self.filename = str(infile)
```

Comment:
Unfortunately, this will not work for Python 2.7 with `u`-strings those contains non-ascii characters.

```python
>>> ConfigObj(u'сюрприз/config.txt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mim/.local/share/virtualenvs/configobj-a6yxznrK/lib/python2.7/site-packages/configobj/__init__.py", line 1210, in __init__
    self._load(infile, configspec)
  File "/home/mim/.local/share/virtualenvs/configobj-a6yxznrK/lib/python2.7/site-packages/configobj/__init__.py", line 1219, in _load
    self.filename = str(infile)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-6: ordinal not in range(128)
```

I've prepared a fix for this situation: mimi1vx/configobj#1